### PR TITLE
*: skip log timeout on maxExecutiontime as slow query for DDL statement.

### LINKED ISSUE #38580: Max execution time mechanism  skip checking DDL jobs
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
Create table and load lots of data; set max_execution_time to a short time; create an index which may cost more than 30 seconds
`
create table t1 (...);
set global max_execution_time = 30000;
create index idx_1 on t1 (col1);
`

Then these error logs will continue to be output until the end of index creation. It's too confusing and misunderstanding.
`
[expensivequery.go:188] [expensive_query] [cost_time=30.021201328s] [conn_id=2430448463150875387] [user=user1] [database=db1] [txn_start_ts=0] [mem_max="0 Bytes (0 Bytes)"] [sql="CREATE INDEX xxxxx ON xxxxx (....)"]"
`

### 2. What did you expect to see? (Required)
Max execution time mechanism  skip checking DDL jobs. There are no DDL jobs to be kill by max_execute_timeout in the logs. 

### 3. What did you see instead (Required)

234dc7de-c747-405e-b694-45e3363bd3ed

### 4. What is your TiDB version? (Required)

`
v6.1.0-20220831
githash: 8473f272ff8372c0a5ffaf5d5d8a787f6ebb9745
`

### PR DESCRIPTION
**Pull request by Benjamin2037**
### What problem does this PR solve?

When user issue an long run DDL statement, it should not be checked in maxExecutionTime for slow query checking mechanism.

Issue Number: close #38580

Problem Summary:
Just Skip none query type statements max execution time checking and kill them.

### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

```release-note
None
```

### CODE REVIEW

File: executor/adapter.go [hunk 1]
```diff
@@ -509,10 +509,14 @@ func (a *ExecStmt) Exec(ctx context.Context) (_ sqlexec.RecordSet, err error) {
 		}
 		maxExecutionTime := getMaxExecutionTime(sctx)
 		// Update processinfo, ShowProcess() will use it.
-		pi.SetProcessInfo(sql, time.Now(), cmd, maxExecutionTime)
 		if a.Ctx.GetSessionVars().StmtCtx.StmtType == "" {
 			a.Ctx.GetSessionVars().StmtCtx.StmtType = ast.GetStmtLabel(a.StmtNode)
 		}
+		// Since maxExecutionTime is used only for query statement, here we limit it affect scope.
```
Comment (xhebox) [inline executor/adapter.go:515]: Maybe https://github.com/pingcap/tidb/blob/58822f885923f60e53831e010f0b215e0b2cff0d/planner/optimize.go#L51 is better alternative for `isQueryStmt`?

Comment (Benjamin2037) [reply executor/adapter.go:515]: Done

**Review by xhebox — COMMENTED** [review]
LGTM, but we do have descriptions:

> Unlike in MySQL, the max_execution_time system variable currently works on all kinds of statements in TiDB, not only restricted to the SELECT statement. The precision of the timeout value is roughly 100ms. This means the statement might not be terminated in accurate milliseconds as you specify.

https://docs.pingcap.com/tidb/dev/system-variables#max_execution_time

This should be investigated.

**Comment by Benjamin2037** [comment]
> LGTM, but we do have descriptions:
> 
> > Unlike in MySQL, the max_execution_time system variable currently works on all kinds of statements in TiDB, not only restricted to the SELECT statement. The precision of the timeout value is roughly 100ms. This means the statement might not be terminated in accurate milliseconds as you specify.
> 
> https://docs.pingcap.com/tidb/dev/system-variables#max_execution_time
> 
> This should be investigated.

This Note is not correct, currently we do not support max_execution_time on all kinds of statements in TiDB.

**Comment by xhebox** [comment]
> > LGTM, but we do have descriptions:
> > > Unlike in MySQL, the max_execution_time system variable currently works on all kinds of statements in TiDB, not only restricted to the SELECT statement. The precision of the timeout value is roughly 100ms. This means the statement might not be terminated in accurate milliseconds as you specify.
> > 
> > 
> > https://docs.pingcap.com/tidb/dev/system-variables#max_execution_time
> > This should be investigated.
> 
> This Note is not correct, currently we do not support max_execution_time on all kinds of statements in TiDB.

https://github.com/pingcap/tidb/pull/10541 According to this PR, I think your argument makes more sense. Then https://github.com/pingcap/docs-cn/blob/master/develop/dev-guide-timeouts-in-tidb.md and its english version needs to be modified along this PR.

**Review by xhebox — APPROVED** [review]

**Pull request merged**